### PR TITLE
BaSP-TG-42: SuperAdmin entity

### PR DESCRIPTION
**Pull request by ProDIGGY-vinilos**
1. List all the super admin

- Retrieve the data from the data base and list all the super admin.
- Create a list view where each item has a button to add, update and delete.

2. Delete an super admin

- Delete a super admin from the list.

3.Update an super admin

- Update a super admin from the list.

### CODE REVIEW

File: src/Components/SuperAdmins/super-admins.module.css [hunk 1]
```diff
@@ -1,6 +1,16 @@
 .container {
   flex: 1;
   display: flex;
+  flex-direction: column;
   justify-content: center;
   align-items: center;
+}
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/super-admins.module.css:8]: Add a space here!

File: src/Components/SuperAdmins/index.js [hunk 2]
```diff
@@ -1,39 +1,48 @@
 import styles from './super-admins.module.css';
 import { useEffect, useState } from 'react';
+// import DeleteButton from './Delete Button/deleteButton';
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/index.js]: delete this comment

File: src/Components/SuperAdmins/super-admins.module.css [hunk 3]
```diff
@@ -4,13 +4,23 @@
   flex-direction: column;
   justify-content: center;
   align-items: center;
+  width: 100%;
 }
-table {
+table{
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/super-admins.module.css:9]: Add a space here!

File: src/Components/SuperAdmins/super-admins.module.css [hunk 4]
```diff
@@ -4,13 +4,23 @@
...
   display: flex;
   flex-direction: column;
-  width: 80%;
+  width: 100%;
+  border: 1px solid #ddd;
+  text-align: center;
 }
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/super-admins.module.css:15]: Add a space here!

File: src/Components/SuperAdmins/super-admins.module.css [hunk 5]
```diff
@@ -4,13 +4,23 @@
...
-#probando{
+th, .info {
+  width: 100%;
+}
+tr{
   display: flex;
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/super-admins.module.css:20]: Add a space here!

File: src/Components/SuperAdmins/super-admins.module.css [hunk 6]
```diff
@@ -4,13 +4,23 @@
...
-  width: 550px;
-}
\ No newline at end of file
+  flex-direction: row;
+  width: 100%;
+}
+.btn{
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/super-admins.module.css:24]: Add a space here!

File: src/Components/SuperAdmins/Form/index.js [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+  const [sAdmin, savesAdmin] = useState({
+    name: '',
+    lastName: '',
+    email: '',
+    password: ''
+  });
+
+  useEffect(async () => {
+    const path = window.location.pathname.split('/');
+    let sAdminId = path[path.length - 1];
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/Form/index.js:15]: you can change "sAdminId" for "**superAdminId**"

File: src/Components/SuperAdmins/Form/form.module.css [hunk 8]
```diff
@@ -0,0 +1,34 @@
+.container{
```
Comment (GonzaloAsencio) [inline src/Components/SuperAdmins/Form/form.module.css:1]: put an empty line between each style.

File: src/Components/SuperAdmins/Form/index.js [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+      } catch (error) {
+        alert(error);
+      }
+    }
+    return;
+  };
+  return (
+    <form className={styles.container}>
+      <div className={styles.header}>
+        <h3>Super admin edit/create</h3>
```
Comment (Ignacio-Sormani) [inline src/Components/SuperAdmins/Form/index.js:73]: You can use a conditional to write 'Super admin edit' or 'Super admin create'

File: src/Components/SuperAdmins/Modal/modal.module.css [hunk 10]
```diff
@@ -0,0 +1,60 @@
...
+    position: fixed;
+    left: 0;
+    top: 0;
+    right: 0;
+    bottom: 0;
+    justify-content: center;
+    align-items: center;
+    background-color: rgba(0,0,0,0.5);
+}
+.modal {
```
Comment (Ignacio-Sormani) [inline src/Components/SuperAdmins/Modal/modal.module.css:11]: Leave one blank line between css blocks

File: src/Components/SuperAdmins/index.js [hunk 11]
```diff
@@ -1,9 +1,50 @@
 import styles from './super-admins.module.css';
+import { useEffect, useState } from 'react';
+import ListSuperAdmin from './ListSuperAdmins/index';
 
 function SuperAdmins() {
+  const [superAdmins, saveSuperAdmins] = useState([]);
```
Comment (Ignacio-Sormani) [inline src/Components/SuperAdmins/index.js:6]: Change the name of the setState function to 'setSuperAdmins'

**Review by Ignacio-Sormani — CHANGES_REQUESTED** [review]

File: src/Components/SuperAdmins/Form/index.js [hunk 12]
```diff
@@ -0,0 +1,99 @@
...
+
+  const onConfirm = (e) => {
+    e.preventDefault();
+    const path = window.location.pathname.split('/');
+    const sAdminId = path[path.length - 1];
+    let url = '';
+    if (sAdminId !== 'super-admin-form') {
+      url = `${process.env.REACT_APP_API_URL}/superAdmins/${sAdminId}`;
+      try {
+        updateDeleteSAdmin('PUT', url);
```
Comment (fede230298) [inline src/Components/SuperAdmins/Form/index.js:56]: Should redirect to SuperAdmin page after a successful request

File: src/Components/SuperAdmins/Form/index.js [hunk 13]
```diff
@@ -0,0 +1,99 @@
...
+      } catch (error) {
+        alert(error.message);
+      }
+    } else {
+      url = `${process.env.REACT_APP_API_URL}/superAdmins`;
+      try {
+        updateDeleteSAdmin('POST', url);
```
Comment (fede230298) [inline src/Components/SuperAdmins/Form/index.js:63]: Same comment as above

File: src/Components/SuperAdmins/ListSuperAdmins/index.js [hunk 14]
```diff
@@ -0,0 +1,51 @@
...
+
+  const closeModal = () => {
+    setModal(false);
+  };
+
+  const deleteSuperAdmin = async (id) => {
+    await fetch(`${process.env.REACT_APP_API_URL}/superAdmins/${id}`, {
+      method: 'DELETE'
+    });
+    onDeleteSuperAdmin(sAdmin._id);
```
Comment (fede230298) [inline src/Components/SuperAdmins/ListSuperAdmins/index.js:21]: Should check the success response before actually deleting the view

**Review by Ignacio-Sormani — APPROVED** [review]

**Review by GonzaloAsencio — APPROVED** [review]

**Review by javierperezraffo — APPROVED** [review]

**Pull request merged**